### ISSUE TITLE
Annotate pdf: fit to screen is problematic with different page sies

### ISSUE THREAD

**Issue by voland66** [xournalpp/xournalpp#2940]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Fedora 33
 - (Linux only) Desktop environment: KDE (X11)
 - Which version of libgtk do you use: 3.24.26
 - Version of Xournal++: 1.1.0+dev (also version provided by fedora)
 - Installation method: rpm built from master branch code (also rpm from Fedora repo) 

**Describe the bug**
Opening pdf file with "Annotate PDF" or "Open" in File menu automatically selects "Zoom fit to screen". If the pdf file has pages of different size, xournal++ attempts to change zoom level when scrolling between different size pages. In the worst case scenario the page on the screen begins to jump back and forth on the screen, constantly re-displaying. It is almost as if the program can't decide which page to fit to the screen. It is difficult to scroll to the next/previous page. When successfully switching to the other page, the document jumps to display the top of that other page. A more tolerable behavior avoids the constant redisplay but the document jumps to the top of the next/previous page.
I am not completely clear what causes the difference between the worst and the more tolerable behavior. It is possible that setting the correct screen dpi solves the worst version of the problem.

Unsetting "Zoom fit to screen" appears to be a workaround. However, opening the pdf file from file menu always turns on "Zoom fit to screen" and I can not find how to disable it in the settings. Opening the pdf file from the command line with "xournalpp filename.pdf" does not set "fit to screen" and problem does not occur. (In fact it appears that "fit to screen" is a default when xournalpp is opened without any arguments.)

**To Reproduce**
1. Create pdf file with pages of different size
2. Open the file for annotation from file menu
3. Scroll between pages of different size.

**Expected behavior**
As one scrolls from page to page the zoom to fit is rescaled at most once without jumping up or down while continuing to display parts of both pages that are supposed to be on the screen.

**Screenshots of Problem**
(Optional) If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other information or context about the problem here.

**Comment by rolandlo** [comment]
I can't reproduce the jumping forth and back. Can you provide a PDF to reproduce it?

**Comment by Rmano** [comment]
Not the same problem, but trying to reproduce:

[mixedsizes.pdf](https://github.com/xournalpp/xournalpp/files/6134499/mixedsizes.pdf)

Open it as `xournalpp mixedsize.pdf` (it opens *without* fit-to-canvas zoom). Then, select page 2, so that also part of page 3 is visible. 

Click "zoom fit-to-canvas".  All is ok. Scroll to the last page, exit the program (no save, since it hasn't changed)

Reopen it as before `xournalpp mixedsize.pdf`. It starts from the last page. Go to page 2, Click zoom-to-canvas and... `xounalpp` sometimes freeze and gives:

```
** (xournalpp:13844): WARNING **: 11:03:06.949: Cannot update zoomPresentationValue yet. This should only happen on startup! 
[1]    13844 segmentation fault (core dumped)  xournalpp mixedsizes.pdf
```

Maybe related, maybe not. Also, sometimes is doesn't happen --- if it doesn't happen the second time you check (the first time never freezes) it will not do it. I suspect that some pesky thing is ending in the "remember last setting" file.

**Comment by rolandlo** [comment]
@Rmano I can reproduce the bug you observe if I open the pdf from the console, scroll to the last page, go into **presentation** mode (not zoom fit mode), exit, reopen and try to scroll up two pages (using the arrow keys).

**Comment by Rmano** [comment]
Hmm. I was quite sure to have hit fit-to-canvas and not presentation mode, but I am not able to reproduce it. Yes, wit presentation mode definitely it happens.

I think that that PDF file is a quite nice stress text.... I'll try to prepare one with big embedded raster image to simulate the horrible PDF scans that my students normally upload (and then slow down xournalpp a lot sometimes).

**Comment by voland66** [comment]
I just realized one thing -- a repeated re-display and jumping around only happen when I scroll with a finger. And it stops when I remove the finger with a somewhat random result -- xournalpp may jump to the top of either of two pages that were on the screen or the page break may remain near the middle of the screen (there is no possibility to have one page occupy, say, 2/3 of the screen vertical space).

If I scroll with a mouse wheel or arrows, there is no repeated jumping around but as the page break comes to about the middle of the screen, xournalpp jumps to the top of the page I am moving towards. 

The problem exists with the mixedsizes.pdf linked by @Rmano  above.

**Comment by rolandlo** [comment]
Should be fixed by #3099
